akka-actor-tests/src/test/scala/akka/NativeImageMetadataSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2009-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import akka.testkit.NativeImageUtils
+import akka.testkit.NativeImageUtils.Constructor
+import akka.testkit.NativeImageUtils.ReflectConfigEntry
+import akka.testkit.NativeImageUtils.ReflectField
+import akka.testkit.NativeImageUtils.ReflectMethod
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+object NativeImageMetadataSpec {
+
+  val metadataDir = NativeImageUtils.metadataDirFor("akka-actor")
+
+  val additionalEntries = Seq(
+    ReflectConfigEntry(
+      classOf[akka.actor.ActorCell].getName,
+      fields = Seq(
+        ReflectField("akka$actor$dungeon$Children$$_childrenRefsDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Children$$_functionRefsDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Children$$_nextNameDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Dispatch$$_mailboxDoNotCallMeDirectly"))),
+    ReflectConfigEntry(
+      classOf[akka.actor.RepointableRef].getName,
+      fields = Seq(ReflectField("_cellDoNotCallMeDirectly"), ReflectField("_lookupDoNotCallMeDirectly"))),
+    // FIXME replace reflective props and drop these
```

Comment:
In progress in #32306

Reply:
Actually, these two are probably not props-by-reflection but config based reflection so will stay here.